scintillometry/io/psrfits/hdu.py

```diff
@@ -170,18 +221,34 @@ def start_time(self):
     def nrow(self):
         return self.header['NAXIS2']
 
+    @nrow.setter
+    def nrow(self, val):
+        self.hdu.header['NAXIS2'] = int(val)
```

Comment:
Here and below, I'd use `operator.index(val)` - we don't want people to pass in floats.